### PR TITLE
Unified debug directory for dynamo/inductor tools

### LINKED ISSUE #1383: Minimizer clobbers preexisting repro.py file without asking
It should do some sort of numeric naming so that old repro files don't get clobbered.

### LINKED ISSUE #1705: minifier is writing repro.py to pytorch install's root directory
This seems totally incorrect, repro.py should be written to the current working directory or to `/tmp`.

```
Writing checkpoint with 4 nodes to /tmp/minifier_soumith/checkpoints/4.py
Copying repro file for convenience to /home/soumith/code/pytorch/repro.py
Wrote minimal repro out to repro.py
```

`/home/soumith/code/pytorch/repro.py` is my PyTorch from-source install that I installed in `python setup.py develop` mode.

My current working directory is `/home/soumith/code/examples/imagenet`

### PR DESCRIPTION
**Pull request by mlazos**
Fixes https://github.com/pytorch/torchdynamo/issues/1705
Fixes https://github.com/pytorch/torchdynamo/issues/1383

Adds a debug directory by default called `torchdynamo_debug` in the current working directory.
In the debug directory for each run of dynamo (an enter and exit of optimize) folder run_\<timestamp\> is created which contains any minifier/inductor/torchdynamo artifacts under respective folders.

Updated the minifier, record replay, and inductor tracing to use this directory

cc @jansel @lezcano @fdrocha @soumith @voznesenskym @yanboliang

### CODE REVIEW

File: test/dynamo/test_replay_record.py [hunk 1]
```diff
@@ -32,14 +31,14 @@ def setUpClass(cls):
         cls._exit_stack.enter_context(
             unittest.mock.patch.object(
                 torch._dynamo.config,
-                "replay_record_dir_name",
-                "/tmp/torch._dynamo_error_records/",
+                "debug_dir_root",
+                "/tmp/_torchdynamo_debug_/",
```
Comment (soumith) [inline test/dynamo/test_replay_record.py:35]: `_torchdynamo_debug_' + os.environ.get( ” USERNAME” )`?

File: torch/_dynamo/debug_utils.py [hunk 2]
```diff
@@ -822,7 +820,7 @@ def debug_wrapper(gm, example_inputs, **kwargs):
                             example_inputs,
                             compiler_name,
                         )
-                    raise ValueError("Issue deteced. Repro at minifier_launcher.py.")
+                    raise ValueError(f"Issue detected. Repro at {get_minifier_repro_path()}.")
```
Comment (soumith) [inline torch/_dynamo/debug_utils.py:823]: while you're at it, can you improve the error message:

```
"Issue detected. You can simplify the issue to a small amount of code by running the minifier with `python {get_minifier_repro_path()} "
```

**Review by soumith — APPROVED** [review]

File: torch/_dynamo/debug_utils.py (hunk 2 above)
Comment (mlazos) [reply torch/_dynamo/debug_utils.py:823]: I'm going to address the minifier printout in a follow up. I haven't totally figured out how the control flow of the minifier works right now, but I get a heavily nested exception printout (this exception nested once for each iteration) so I'm going to fix the minifier printouts entirely and have a status message on each iteration.

File: test/dynamo/test_replay_record.py (hunk 1 above)
Comment (mlazos) [reply test/dynamo/test_replay_record.py:35]: This is just test code which gets cleaned up afterward, so I'll leave this if you're okay with it.

File: torch/_dynamo/debug_utils.py (hunk 2 above)
Comment (soumith) [reply torch/_dynamo/debug_utils.py]: sounds good

File: test/dynamo/test_replay_record.py (hunk 1 above)
Comment (soumith) [reply test/dynamo/test_replay_record.py:35]: sounds good

Comment (yanboliang) [reply test/dynamo/test_replay_record.py:35]: I'd prefer to use ```+ USERNAME``` as many users working on share cluster like us, other folks may pollute or overwrite your debug or tracing output.

**Pull request closed without merging**